File: generator/src/test/resources/swift/expected/ProblemTypesTest/generates-problem-types.output3.swift
import Foundation
import Sunday

public class TestResolverProblem : Problem {

  public static let type: URL = URL(string: "http://example.com/test_resolver")!
  public let optionalString: String?
  public let arrayOfStrings: [String]
  public let optionalArrayOfStrings: [String]?
  public override var description: String {
    return DescriptionBuilder(Self.self)
        .add(type, named: "type")
        .add(title, named: "title")
        .add(status, named: "status")
        .add(detail, named: "detail")
        .add(instance, named: "instance")
        .add(optionalString, named: "optionalString")
        .add(arrayOfStrings, named: "arrayOfStrings")
        .add(optionalArrayOfStrings, named: "optionalArrayOfStrings")
        .build()
  }

  public init(
    optionalString: String? = nil,
    arrayOfStrings: [String],
    optionalArrayOfStrings: [String]? = nil,
    instance: URL? = nil
  ) {
    self.optionalString = optionalString
    self.arrayOfStrings = arrayOfStrings
    self.optionalArrayOfStrings = optionalArrayOfStrings
    super.init(type: Self.type, title: "Test Resolve Type Reference", status: 500,
        detail: "Tests the resolveTypeReference function implementation.", instance: instance,
        parameters: nil)
  }

  public required init(from decoder: Decoder) throws {
    let container = try decoder.container(keyedBy: CodingKeys.self)
    self.optionalString = try container.decodeIfPresent(String.self, forKey: CodingKeys.optionalString)
    self.arrayOfStrings = try container.decode([String].self, forKey: CodingKeys.arrayOfStrings)
    self.optionalArrayOfStrings = try container.decodeIfPresent([String].self, forKey: CodingKeys.optionalArrayOfStrings)
    try super.init(from: decoder)
  }

  public override func encode(to encoder: Encoder) throws {
    try super.encode(to: encoder)
    var container = encoder.container(keyedBy: CodingKeys.self)
    try container.encode(self.optionalString, forKey: CodingKeys.optionalString)
    try container.encode(self.arrayOfStrings, forKey: CodingKeys.arrayOfStrings)
    try container.encode(self.optionalArrayOfStrings, forKey: CodingKeys.optionalArrayOfStrings)
  }

  fileprivate enum CodingKeys : String, CodingKey {

    case optionalString = "optionalString"
    case arrayOfStrings = "arrayOfStrings"
    case optionalArrayOfStrings = "optionalArrayOfStrings"

  }

}

File: generator/src/test/resources/kotlin/expected/ProblemTypesTest/generates-problem-types.output3.kt
package io.test.service

import com.fasterxml.jackson.`annotation`.JsonIgnore
import java.net.URI
import kotlin.String
import kotlin.collections.List
import org.zalando.problem.AbstractThrowableProblem
import org.zalando.problem.Exceptional
import org.zalando.problem.Status
import org.zalando.problem.ThrowableProblem

public class TestResolverProblem(
  public val optionalString: String? = null,
  public val arrayOfStrings: List<String>,
  public val optionalArrayOfStrings: List<String>? = null,
  instance: URI? = null,
  cause: ThrowableProblem? = null,
) : AbstractThrowableProblem(TYPE_URI, "Test Resolve Type Reference", Status.INTERNAL_SERVER_ERROR,
    "Tests the resolveTypeReference function implementation.", instance, cause) {
  @JsonIgnore
  override fun getCause(): Exceptional? = super.cause

  public companion object {
    public const val TYPE: String = "http://example.com/test_resolver"

    public val TYPE_URI: URI = URI(TYPE)
  }
}
